.github/workflows/ci.yml

```diff
@@ -1,32 +1,64 @@
 name: CI
 
-on: [push, pull_request]
+on: 
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
 
 jobs:
   build:
     runs-on: ubuntu-latest
-    services:
-      postgres:
-        image: postgres:11
-        env:
-          POSTGRES_USER: postgres
-          POSTGRES_PASSWORD: postgres
-          POSTGRES_DB: ci_db_test
-        ports:
-          - 5432:5432
-        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
     strategy:
       matrix:
-        node: ['8', '10', '12', '14', '16', '17']
-        os: [ubuntu-latest, windows-latest, macos-latest]
+        #node: ['8', '10', '12', '14', '16', '17']
+        #os: [ubuntu-latest, windows-latest, macos-latest]
+        # let's make it a little bit simple for now
+        node: ['17']
+        os: [ubuntu-latest]
     name: Node.js ${{ matrix.node }} (${{ matrix.os }})
     steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
+      
       - name: Setup node
         uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node }}
           cache: yarn
-      - run: yarn install
-      # TODO(bmc): get ssl tests working in ci
-      - run: PGTESTNOSSL=true PGUSER=postgres PGPASSWORD=postgres PGDATABASE=ci_db_test yarn test
+
+      - name: build
+        run: yarn 
+
+      - name: boostrap
+        run: yarn lerna bootstrap
+        
+      - name: Setup Vertica
+        run: |          
+          docker pull vertica/vertica-ce:12.0.0-0
+          docker run -d -p 5433:5433 -p 5444:5444 \
```

Comment:
This takes some time to startup and configure the DB.  Do we need to add a sleep after docker run?  This may be related to the current test failures.

If you manually perform this docker run locally you can see what an appropriate sleep might be (30s, 60s, etc, of course it may need to be different in GitHub).

Reply:
If the tests still fail when accessing localhost, another option is to use the service name as the hostname (`V_HOST=vertica_ce` in this case).

Reply:
yep, I think the reason the v-protocol package didn't pass before is due to Vertica not up totally, I have added a vsql probe to make sure Vertica startup when these tests running